lib/streamlit/commands/query_params.py

```diff
@@ -143,29 +143,102 @@ def _ensure_no_embed_params(
     return current_embed_params
 
 
+def _missing_key_error_message(key: str) -> str:
+    return f'st.query_params has no key "{key}". Did you forget to initialize it? '
+
+
 @dataclass
 class QueryParams:
     """A dict-like representation of query params.
-    The main difference is that it only stores and returns str and list[str].
+    The main difference is that it only stores and returns str and List[str].
 
     TODO(willhuang1997): Fill in these docs with examples and fix doc above. Above is just a stub for now.
     """
 
-    _query_params: Dict[str, List[Any]] = field(default_factory=dict)
+    _query_params: Dict[str, Union[List[str], str]] = field(default_factory=dict)
+
+    def __init__(self, query_params: Dict[str, Union[List[str], str]] = {}):
+        # avoid using ._query_params as that will use __setattr__
+        # and access _query_params which is None as _query_params isn't initialized
+        self.__dict__["_query_params"] = query_params
+
+    def _keys(self) -> Set[str]:
+        return set(self._query_params.keys())
```

Comment:
Is this helper method used anywhere else? We might as well just get rid of it if it's only used to define `__iter__`. We can also probably just define `__iter__` as  `return iter(self._query_params.keys())`